packages/origin-ui/src/features/contracts/sagas.ts

```diff
@@ -199,17 +193,20 @@ function* initEventHandler() {
     }
 }
 
-function* fillOriginContractLookupAddressIfMissing(): SagaIterator {
-    yield takeEvery(ContractsActions.setOriginContractLookupAddress, function*() {
-        let originContractLookupAddress: string = yield select(getOriginContractLookupAddress);
+function* fillMarketContractLookupAddressIfMissing(): SagaIterator {
+    yield takeEvery(ContractsActions.setMarketContractLookupAddress, function*() {
+        let marketContractLookupAddress: string | undefined = yield select(
+            getMarketContractLookupAddress
+        );
+
         const loading: boolean = yield select(getLoading);
 
-        if (originContractLookupAddress && loading) {
+        if (marketContractLookupAddress && marketContractLookupAddress !== 'undefined' && loading) {
```

Comment:
`marketContractLookupAddress !== 'undefined'` already covers `marketContractLookupAddress` truthiness check, so `marketContractLookupAddress && ` can be removed